BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.java

```diff
@@ -126,42 +136,84 @@ public BraintreeClient(@NonNull Context context, @NonNull String authorization,
      *
      * @param callback {@link ConfigurationCallback}
      */
-    public void getConfiguration(@NonNull ConfigurationCallback callback) {
-        configurationLoader.loadConfiguration(applicationContext, authorization, callback);
+    public void getConfiguration(@NonNull final ConfigurationCallback callback) {
+        getAuthorization(new AuthorizationCallback() {
+            @Override
+            public void onAuthorizationResult(@Nullable Authorization authorization, @Nullable Exception error) {
+                if (authorization != null) {
+                    configurationLoader.loadConfiguration(applicationContext, authorization, callback);
+                } else {
+                    callback.onResult(null, error);
+                }
+            }
+        });
+    }
+
+    void getAuthorization(@NonNull final AuthorizationCallback callback) {
+        authorizationLoader.loadAuthorization(callback);
+    }
+
+    AuthorizationType getAuthorizationType() {
+        return authorizationLoader.getAuthorizationType();
     }
```

Comment:
This allows the `AuthorizationType` to be inferred synchronously without triggering a client token fetch e.g. TokenizationKey or ClientToken.